src/App.js

```diff
@@ -12,134 +12,135 @@ import imgLoading from "./images/isLoading.gif";
 
 class App extends React.Component {
   state = {
-    todos: [],
+    todos: {
+      0: {
+        id: 0,
+        content: "sell a keyboard",
+        done: false,
+      },
+      1: {
+        id: 1,
+        content: "buy a mouse",
+        done: false,
+      },
+      2: {
+        id: 2,
+        content: "play a new game",
+        done: true,
+      },
+    },
     status: "ALL",
     isLoading: false,
     error: "",
   };
 
-  componentDidMount() {
-    this.setState({
-      isLoading: true,
-    });
-    axios
-      .get(URL.TODOS)
-      .then((response) => {
-        this.setState({
-          todos: response.data,
-        });
-      })
-      .catch((err) => {
-        this.setState({
-          error: err.data,
-        });
-      })
-      .finally(() => {
-        this.setState({
-          isLoading: false,
-        });
-      });
-  }
-
   addTodo = (todo) => {
-    const newTodo = {
-      id: Date.now(),
-      content: todo.content,
-      done: false,
+    const id = Date.now();
+
+    const newTodos = {
+      [id]: {
+        id,
+        content: todo.content,
+        done: false,
+      },
     };
-    axios.post(URL.TODOS, newTodo).then(() => {
-      this.setState({
-        todos: this.state.todos.concat(newTodo),
-      });
+    this.setState({
+      todos: Object.assign({}, this.state.todos, newTodos),
     });
   };
 
   toggleAll = () => {
     const { todos } = this.state;
-    const completedTodoNumber = todos.filter((todo) => todo.done).length;
-    const willSetToTrue = completedTodoNumber !== todos.length;
+    const countDone = Object.keys(todos)
+      .map((key) => todos[key])
+      .filter((todo) => todo.done).length;
 
-    const todoTobeUpdated = todos.filter(
-      (todo) => todo.done === !willSetToTrue
-    );
+    const doneValue = Object.keys(todos).length !== countDone;
 
-    for (let index = 0; index < todoTobeUpdated.length; index++) {
-      const todo = todoTobeUpdated[index];
-      axios.put(`${URL.TODOS}/${todo.id}`, {
-        content: todo.content,
-        done: !todo.done,
+    const newTodos = Object.keys(todos).reduce((acc, key) => {
+      return Object.assign({}, acc, {
+        [key]: { ...todos[key], done: doneValue },
       });
-    }
+    }, {});
 
     this.setState({
-      todos: todos.map((todo) => ({
-        ...todo,
-        done: willSetToTrue,
-      })),
+      todos: newTodos,
     });
   };
 
-  toggleTodo = (todo, id) => {
-    axios
-      .put(`${URL.TODOS}/${id}`, {
-        content: todo.content,
-        done: !todo.done,
-      })
-      .then(() => {
-        this.setState({
-          todos: this.state.todos.map((todo) => ({
-            ...todo,
-            done: todo.id === id ? !todo.done : todo.done,
-          })),
-        });
-      });
+  toggleTodo = (id, content, done) => {
```

Comment:
why doesn't this change to object like editTodo?
Did I tell u abt this? How about other places?

Reply:
i did